CMakeLists.txt

```diff
@@ -1,57 +1,81 @@
 cmake_minimum_required(VERSION 3.14)
 project(iggy-cpp-client)
 
-set(CMAKE_CXX_STANDARD 14)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
+option(ENABLE_GTEST "Enable GoogleTest" OFF)
+option(ENABLE_DOXYGEN "Enable Doxygen" OFF)
 
 # avoid warning about DOWNLOAD_EXTRACT_TIMESTAMP in CMake 3.24
 if (CMAKE_VERSION VERSION_GREATER_EQUAL "3.24.0")
   cmake_policy(SET CMP0135 NEW)
 endif()
 
-include(FetchContent)
-FetchContent_Declare(
-  googletest
-  URL https://github.com/google/googletest/archive/dddb219c3eb96d7f9200f09b0a381f016e6b4562.zip
-)
-set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
-FetchContent_MakeAvailable(googletest)
+# set up library dependencies
+find_package(argh CONFIG REQUIRED)
+find_package(benchmark CONFIG REQUIRED)
+find_package(libuv CONFIG REQUIRED)
+find_package(spdlog CONFIG REQUIRED)
 
 add_library(
   iggy
 
-  sdk/message.cc
+  sdk/models.cc
 )
-
-# set up GoogleTest
-enable_testing()
+target_compile_features(iggy PRIVATE cxx_std_17)
 
 add_executable(
-  iggy_cpp_test
+  iggy_cmd
 
-  tests/message_test.cc
+  sdk/iggy_cmd.cc
 )
+target_include_directories(iggy_cmd PRIVATE ${argh_INCLUDE_DIR})
 target_link_libraries(
-  iggy_cpp_test
+  iggy_cmd
 
   iggy
-  GTest::gtest_main
 )
 
-include(GoogleTest)
-gtest_discover_tests(iggy_cpp_test)
+# set up GoogleTest
+if(ENABLE_GTEST)
```

Comment:
I suggest moving this to separate file, include is via 
`add_subdirectory` and call fetch content there

also, IMHO, dont the the variable `ENABLE_GTEST`, because the main purpose of this is not to enable google test framework, but to build tests :D

so I suggest:
`BUILD_TESTS`, `UT` or `UNIT_TESTS`

Reply:
To be resolved in follow-up issue #12.